assets/js/script.js

```diff
@@ -46,14 +49,62 @@ jQuery( function ( $ ) {
 			const value = $( this ).val().toLowerCase();
 
 			$( '.file-upload-types-table table tr' ).filter( function () {
-				if ( ! $( this ).hasClass( 'heading' ) ) {
-					if ( value !== '' && $( this ).hasClass( 'section' ) ) {
-						$( this ).hide();
+
+				const $this = $( this );
+
+				if ( ! $this.hasClass( 'heading' ) ) {
+					if ( value !== '' && $this.hasClass( 'section' ) ) {
+						$this.hide();
 					}
-					else {
-						$( this ).toggle( $( this ).text().toLowerCase().indexOf( value ) > - 1 );
+					// Do nothing of this is hidden repetitive section with no value yet.
+					else if ( $this.hasClass( 'repetitive-fields' ) && $this.css( 'display' ) === 'none' && ! $this.find( 'input' ).toArray().some( input => input.value !== '' ) ) {
+						return;
+					} else {
+						$this.toggle( $this.text().toLowerCase().indexOf( value ) > - 1 );
 					}
 				}
 			} );
+		} )
+		.on( 'click', '#c_types_file_sample_button a', function ( e ) {
+			e.preventDefault();
+
+			$( '.repetitive-fields' ).show();
 		} );
+
+	let uploaded = 0;
+
+	let SampleFileDropzone = new Dropzone( '#c_types_file_sample_button', {
+		url: ajaxurl,
+		uploadMultiple: false,
+		allowMultiple: false,
+		previewTemplate: '<div style="display:none"></div>',
+		init: function() {
+			this.on( 'sending', function( file, xhr, formData ) {
+				formData.append( 'action', 'file_upload_types_check_sample' );
+				formData.append( 'nonce', file_upload_types_params.nonce );
+
+				$( '.file-upload-types-dropzone span.icon' ).addClass( 'loading' );
+			} );
+			this.on( 'success', function( file, response ) {
```

Comment:
What do you think if we cover an edge case when we can't detect mime type or extension?

I would return the 400th error and add the `on( 'error', ...` callback which shows an error message for customers.

Reply:
done